### ISSUE TITLE
Error with react native

### ISSUE THREAD

**Issue by shtanton** [calmm-js/partial.lenses#161]
I tried to use this package on a react native package but got this error:
`In this environment the sources for assign MUST be an object.  This error is a performance optimization and not spec compliant`
I did some digging and found what it is complaining about:
`src/ext/infestines.js:12`
`export const protoless0 = I.freeze(protoless(0))`
This calls the protoless function which assigns the value 0 to an empty object, which is what throws the error.
I am wondering why as I couldn't find any changes to the object after the 0 is assigned and if I replace `protoless(0)` with `create(null)` and run the tests, everything seemed to work.

Maybe there is a reason for this that I am missing?  Thanks in advance for any help

**Comment by polytypic** [comment]
Hmm... I believe Stefan Rimaila (@stuf) [ran into this same issue](https://gitter.im/calmm-js/chat?at=5a7997ddce68c3bc7450039c) some time ago.

Would you like to make a quick PR with the following diff

```diff
-export const protoless0 = I.freeze(protoless(0))
+export const protoless0 = I.freeze(protoless(I.object0))
```

which should also work around the issue with RN?  I'm hesitant to make more significant workarounds for RN, but if the immediate crash is due to just that one line, then why not just change it.

The root problem is that, for some reason, React Native implements a polyfill (or a replacement?) for the standard [`Object.assign`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) function that is not [spec](https://www.ecma-international.org/ecma-262/6.0/#sec-object.assign) compliant.  In other words, it doesn't work like the JavaScript standard specifies it should work.

Partial Lenses basically uses `Object.assign` internally in some object manipulating optics to convert objects with non-trivial prototypes to plain objects that can be iterated over safely using `for (const k in o)`.  It is also possible that Partial Lenses calls `Object.assign` with arguments that are not objects, which is something that `Object.assign` should also handle (according to [spec](https://www.ecma-international.org/ecma-262/6.0/#sec-object.assign) it converts `null` and `undefined` to (effectively) empty objects and calls `ToObject` on other inputs).

I mention the above because, even though the change to `protoless0` might make Partial Lenses to not immediately crash on React Native, it is still the case that some operations with Partial Lenses might crash on React Native due to the non-spec compliant `Object.assign` implementation on that particular platform.  This might happen, for example, when you try to access some value `x` that is not an object with an object manipulating optic.  Normally such an operation would treat the value `x` in the same way as if it was `undefined`, but with RN the operation would instead throw an error due to the non-spec compliant `Object.assign`.

**Comment by shtanton** [comment]
Done
[Pull Request](https://github.com/calmm-js/partial.lenses/pull/162/commits)

**Comment by polytypic** [comment]
Thanks for reporting this and the PR!  I've now released v13.7.4 that contains the workaround.
